### PR TITLE
Reader Tracks: add tracking to clicking comment timestamp (permalink)

### PR DESCRIPTION
**Pull request by aaronyan**
Adds tracking to the the comment timestamp (a permalink to the front-end site's comment). Also adjusts the stats method `recordPermalinkClick()` to be more flexible to non-post tracking.

### CODE REVIEW

**Comment by aaronyan** [comment]
Attempts to solve #18585

File: client/reader/stats.js [hunk 1]
```diff
@@ -22,20 +22,25 @@ export function recordGaEvent( action, label, value ) {
 	ga.recordEvent( 'Reader', action, label, value );
 }
 
-export function recordPermalinkClick( where, post ) {
+export function recordPermalinkClick( where, post, eventProperties = {} ) {
```
Comment (samouri) [inline client/reader/stats.js]: sounds like `where` should be named `source` 🤔

File: client/reader/stats.js [hunk 2]
```diff
@@ -22,20 +22,25 @@ export function recordGaEvent( action, label, value ) {
...
 	const trackEvent = 'calypso_reader_permalink_click';
-	const args = {
-		source: where,
-	};
+
+	// Add where as source prop
+	eventProperties = Object.assign( { source: where }, eventProperties );
+
+	// Add location as ui_algo prop
+	const location = getLocation();
```
Comment (samouri) [inline client/reader/stats.js]: do you need to take care of `ui_algo` here since `recordTrack` will do it for you?

Comment (aaronyan) [reply client/reader/stats.js]: Ah yes! Good point. Removing!

**Comment by aaronyan** [comment]
Updated @samouri! Thanks for the 👀. Looks good to go?

File: client/reader/stats.js [hunk 3]
```diff
@@ -22,20 +22,21 @@ export function recordGaEvent( action, label, value ) {
...
 	} );
-	recordGaEvent( 'Clicked Post Permalink', where );
+	recordGaEvent( 'Clicked Post Permalink', source );
 	const trackEvent = 'calypso_reader_permalink_click';
-	const args = {
-		source: where,
-	};
+
+	// Add source as Tracks event property
+	eventProperties = Object.assign( { source: source }, eventProperties );
```
Comment (samouri) [inline client/reader/stats.js]: nit: you can now write `{ source }`  instead of `{ source: source }`.

**Pull request merged**